portfolio/src/main/webapp/script.js

```diff
@@ -13,34 +13,28 @@
 // limitations under the License.
 
 
-/** @return {string} */
-function generateRandomGreeting(){
+/**
+ * @return {string} 
+ * a random greeting
+ */
+function generateRandomGreeting() {
     //TODO use google translated to get greetings in different languages
     const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+        ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
     // Pick a random greeting.
-    const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    var greeting = greetings[Math.floor(Math.random() * greetings.length)];
     return greeting;
 }
 
 /**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = generateRandomGreeting();
-}
-
-/**
- * Change the innerHTML to a greeting
+ * Change the innerHTML to a greeting and name
  * every time use loads or refreshes
  */
-function onLoadWindow(){
-    document.getElementById("welcome-note").innerHTML = generateRandomGreeting().concat(" My name is Roland");
+function loadName() {
+    fetch("/data").then(response => response.json()).then((user) => {
```

Comment:
I think you could simplify this .then chain a bit
.then(response => response.json()) returns response.json to the next then statement, but since response.json() is just reading a value in the response, you can do it all in one "then" function:

fetch("/data").then( (response) => {
const user = response.json();
document.getElementById(.......) // rest of your code here
});

Let me know if this comment doesn't make sense, kind of hard to talk about then chains over text, happy to chat in person too :)

Reply:
Building on this a bit, you only want to chain an additional `.then()` on a promise.  In this case, Vita's suggesting not chaining a `.then()` when you retrieve `response.json` because `response.json` is not a promise

Reply:
I couldn't get to work. However, when I hover over the response.json() looks like it is a promise, but it might also be because of the implementation.
definitely would love to talk in person, that would help a lot